diff -Naur drakguard-0.7.7/bin/drakguard drakguard-0.7.7.tpg/bin/drakguard
--- drakguard-0.7.7/bin/drakguard	2010-05-25 12:20:14.000000000 +0000
+++ drakguard-0.7.7.tpg/bin/drakguard	2014-03-27 19:33:35.149867411 +0000
@@ -432,7 +432,7 @@
 		 join('', map {
 		     my $chain = $_;
 		     map {
-			 sprintf("iptables -I $chain -j DROP -m time --timestart %02d:%02d --timestop %02d:%02d\n",
+			 sprintf("iptables -I $chain -j DROP -m time --timestart %02d:%02d --timestop %02d:%02d --kerneltz\n",
 				 int($_->[0]/60), $_->[0]%60,
 				 int($_->[1]/60), $_->[1]%60,
 			     );
