diff -Naur drakguard-0.7.7/bin/drakguard drakguard-0.7.7.tpg/bin/drakguard
--- drakguard-0.7.7/bin/drakguard	2010-05-25 12:20:14.000000000 +0000
+++ drakguard-0.7.7.tpg/bin/drakguard	2014-03-27 19:46:16.628193716 +0000
@@ -294,27 +294,28 @@
 );
 
 $w->show;
-if ( (grep {$_ && !/acl/} chomp_(cat_($fstab_file)) && grep {$_ && m/ext/} chomp_(cat_($fstab_file)))) {
+# FIXME: bogus check: if ACLs are disabled everywhere but in /unused/path, it's OK...
+if (cat_('/boot/config') =~ /_FS_POSIX_ACL=y/ &&  grep { $_ && !/acl/ && !/^#/ && m/ext[23]|reiserfs/ } chomp_(cat_($fstab_file))) {
     $acl_active = 1;
 	$in->ask_yesorno(N("Warning"), N("The support for Access Control Lists must be enabled in order to use the Block programs feature.\nDo you want to enable it now?", $acl_active = 1)) or $acl_active = 0;
-    if ($acl_active) { 
-	my @ext_lines = grep {$_ && m/ext/} chomp_(cat_($fstab_file));
+    if ($acl_active) {
+	my @ext_lines = grep { $_ && m/ext[23]|reiserfs/ && !/^#/ } chomp_(cat_($fstab_file));
     	foreach my $line (@ext_lines){
-            my ($id, $directory, $flag, $number) = $line=~ /^(.+)\s(.+)\s(ext(\w)\s(\w*))\s(.*)/ or next;
+            my ($flag) = $line =~ /^.+\s.+\s(ext(\w)\s(\w*))\s/ or next;
             $line =~ s/ext(\w)\s(\w*)\s/$flag,acl /g;
-            subst_config_line($fstab_file,"$line\n"); #adds ,acl in the fstab file in the first run
+            subst_config_line($fstab_file, "$line\n"); #adds ,acl in the fstab file in the first run
             # remount filesystem with acl parameter
     	}
 	$acl_active = 0;
-    $in->ask_warn(N("Warning"), N("The support for Access Control Lists was enabled.") . N("It is necessary to restart your computer to activate it."));
+    $in->ask_warn(N("Warning"), N("The support for Access Control Lists was enabled.") . "\n" . N("It is necessary to restart your computer to activate it."));
     }
 }
 else {
     my @mount = `mount`;
-    if (grep (/,acl/, @mount)) {
-        $acl_active = 1;
+    if (grep { !/,acl/ && m/ext[23]|reiserfs/ } @mount) {
+	$in->ask_warn(N("Warning"), N("The support for Access Control Lists, required by the Block programs feature, is enabled, but not yet activated.") . N("It is necessary to restart your computer to activate it."), $acl_active = 0);
     } else {
-        $in->ask_warn(N("Warning"), N("The support for Access Control Lists, required by the Block programs feature, is enabled, but not yet activated.") . N("It is necessary to restart your computer to activate it."), $acl_active = 0);}
+    $acl_active = 1 }
 }
 update_time_change;
 Gtk2->main;
