--- drakguard/bin/drakguard	2013/06/30 22:15:12	8582
+++ drakguard/bin/drakguard	2013/07/02 15:37:34	8583
@@ -69,6 +69,7 @@
 my $proxy_user = 'squid';
 my $guardian_port = 8080;
 my $guardian_user = 'dansguardian';
+my $root_user = 'root';
 load();
 
 
@@ -91,7 +92,7 @@
 
 my $users_list = Gtk2::SimpleList->new(N("Allowed users") => 'text');
 $users_list->get_selection->set_mode('multiple');
-@{$users_list->{data}} = difference2($shorewall->{accept_local_users}{http}, [ $proxy_user ]);
+@{$users_list->{data}} = difference2($shorewall->{accept_local_users}{http}, [ $root_user, $proxy_user ]);
 
 my @url_lists = (
     {
@@ -466,7 +467,7 @@
 
     if ($shorewall) {
         $shorewall->{disabled} = 0 if $enable;
-        @{$shorewall->{accept_local_users}{http}} = if_($enable, uniq($proxy_user, list_get_entries($users_list)));
+        @{$shorewall->{accept_local_users}{http}} = if_($enable, uniq($proxy_user, $root_user, list_get_entries($users_list)));
         @{$shorewall->{accept_local_users}{$proxy_port}} = if_($enable, $guardian_user);
         network::shorewall::set_redirected_ports($shorewall, 'tcp', $guardian_port, if_($enable, 'http', $proxy_port));
         network::shorewall::write($shorewall, $in);
